server/errors/errors.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package errors
+
+import (
+	te "github.com/google/trillian/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+)
+
+// WrapError wraps err as a GRPC error if err is a TrillianError, else err is
+// returned unmodified.
+func WrapError(err error) error {
+	switch err := err.(type) {
+	case te.TrillianError:
+		return grpc.Errorf(codes.Code(err.Code()), err.Error())
```

Comment:
The 1:1 mapping assumption is very much on purpose. While we *could* break it, I don't see *why* we'd want that. The error set we get from GRPC is pretty comprehensive (I actually doubt we'll use more than a handful of the errors there). There's the potential for a distinct error space, as you said, but so far this is isn't it.

For the sake of argument, in case we went that route, we'd simply remove the 1:1 assumption from this test and push the mapping responsibility to the translation layer (ie, trillian/server/errors). The mapping would become n:1 (trillian.errors.Code to grpc.codes.Code). We don't ever map the other way around, so that would not be a problem.